tcmur_cmd_handler.c

```diff
@@ -144,6 +144,124 @@ static int write_work_fn(struct tcmu_device *dev,
 				block_size * tcmu_get_lba(cmd->cdb));
 }
 
+static void handle_write_same_16_null_cbk(struct tcmu_device *dev,
+					  struct tcmulib_cmd *cmd,
+					  int ret)
+{
+}
+
+static int write_same_16_work_fn(struct tcmu_device *dev,
+				 struct tcmulib_cmd *cmd)
+{
+	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
+	uint32_t block_size = tcmu_get_dev_block_size(dev);
+	uint32_t lba_cnt = tcmu_get_xfer_length(cmd->cdb);
+	uint64_t start_lba = tcmu_get_lba(cmd->cdb);
+	uint64_t end_lba;
+	uint32_t max_lbas, copy_lbas;
+	uint32_t max_sectors;
+	int ret;
+
+	end_lba = start_lba + lba_cnt;
+
+	max_sectors = tcmu_get_attribute(dev, "hw_max_sectors");
```

Comment:
Shoot. What is hw_max_sectors? In tcmu_emulate_evpd_inquiry, it looks like the max number of 512 byte sectors.

I think  tcmu_emulate_evpd_inquiry is wrong. From the kernel target_core_iblock use  it looks like it is supposed to be max number of block_size sectors, right? So your code is right and we need to fix  tcmu_emulate_evpd_inquiry.